CONTRIBUTING.md

```diff
@@ -0,0 +1,210 @@
+because X11 has a lot of issues and inconsistencies especially with regards to
+pen input which is a integral part of the application.
+
+For more details how to build the application on linux either natively or as flatpak see: [BUILDING.md](./BUILDING.md).
+
+## MacOS
+
+The application is also bundled for MacOS, @dehesselle is active in issues that affect the app bundle.
+
+For more details how to build the application on MacOS
+see: [rnote-macos-build.md](./misc/building/rnote-macos-build.md).
+
+## Windows
+
+For windows `mingw64` is used as the development and build environment.
+For the installer `innosetup` is used.
+
+It should always be ensured that the app will build in `mingw64`,
+but it's not a focus to ensure tight integration with Windows OS.
+
+For more details how to build the application and the installer on Windows
+see: [rnote-windows-build.md](./misc/building/rnote-windows-build.md).
+
+# Translations
+
+A great way to contribute to the project without writing code is adding a new
+or start maintaining an existing translation language.
+The translations files are located in `crates/rnote-ui/po/`.
+
+ Creating translations for new languages or updating existing ones can be done in multiple ways:
+- take the `rnote.pot` file and generate a new `.po` translation file from it, for example with "Poedit".
+    Add the new translation language to `LINGUAS` and submit a PR with the changed files.
+- use [weblate](https://hosted.weblate.org/projects/rnote/repo/) for an easy way to translate in the browser
+    without having to deal with git.
+
+# Code Style
+
+## Formatting
+
+For formatting `rustfmt` is used. It picks up the formatting configuration file `rustfmt.toml`.
+
+To check the formatting run:
+
+```bash
+cargo fmt --check
+```
+
+And to directly apply
+
+```bash
+cargo fmt
+```
+The formatting is also checked in the CI and a prerequisite for merging additional or changed code. 
+
+## Lints
+
+For linting `clippy` is used.
+Because the app needs to be built through meson, there is a meson target the print lints available:
+
+For the UI
+
+```bash
+meson compile ui-cargo-clippy -C _mesonbuild
+```
+
+For the CLI
+
+```bash
+meson compile cli-cargo-clippy -C _mesonbuild
+```
+
+If consciously considered clippy warnings can also disabled in code by using `#[allow(clippy::<lint-name>)].
+However this must be justified when getting new code in.
+
+## Pre-Commit hooks
+
+Per default on an initial build git pre-commit hooks are installed to ensure the outlined style consistency.
+Check out [pre-commit.hook](hooks/pre-commit.hook) to see what the hook will do in detail.
+
+# Tests
+
+## Unit Tests
+
+Some unit tests are specified throughout the codebase.
+
+To run them, execute:
+
+```bash
+meson compile cargo-test -C _mesonbuild
+```
+
+Adding tests can be done like in any other rust crate by adding `tests` modules
+and `#[test]` cases directly in code in the same files where the code that should be tested is written.
```

Comment:
Suggestion: 

"Just like in any other rust crate, tests can be added by declaring a `tests` module prefixed with the `#[cfg(test)]` attribute, then adding test functions prefixed by the `#[test]` attribute."

"Tests should be as closely coupled to the code they target as reasonably possible."